core/src/main/scala/com/softwaremill/sttp/testing/SttpBackendStub.scala

```diff
@@ -159,13 +164,19 @@ object SttpBackendStub {
                                PartialFunction.empty,
                                Some(fallback))
 
-  private[sttp] def tryAdjustResponseType[T, U](ra: ResponseAs[T, _],
-                                                r: Response[U]): Response[_] = {
-    r.body match {
-      case Left(_) => r
-      case Right(body) =>
-        val newBody: Any = tryAdjustResponseBody(ra, body).getOrElse(body)
-        r.copy(body = Right(newBody))
+  private[sttp] def tryAdjustResponseType[DesiredRType, RType, M[_]](
+      rm: MonadError[M],
+      ra: ResponseAs[DesiredRType, _],
+      m: M[Response[RType]]): M[Response[DesiredRType]] = {
+    rm.map[Response[RType], Response[DesiredRType]](m) { r =>
```

Comment:
[minor] probably could be a bit shorter by just saying `.map { _.body match { ... `

Reply:
How? We use `r` in both match cases.

Reply:
True :)